### PR TITLE
Initialize Jupyter Book documentation

### PR DESCRIPTION
**Pull request by weiji14**
## What I am changing

- Documentation website built with [Jupyter-Book](https://github.com/executablebooks/jupyter-book/tree/v0.15.1)/Sphinx.
- **Preview** at https://clay-foundation.github.io/model/intro.html
- Screenshot:
image

## How I did it

- Follow instructions at https://jupyterbook.org/en/stable/start/create.html
- Deployed to GitHub Pages via GitHub Actions - https://jupyterbook.org/en/stable/publish/gh-pages.html#automatically-host-your-book-with-github-actions
- Based on https://github.com/geo-smart/deepicedrainpipe/pull/2

TODO in this PR:
- [x] Add jupyter-book dependency
- [x] Initialize minimal Jupyter-Book site
- [x] Setup GitHub Actions CI to build docs and push to GitHub Pages
- [x] Add more sections on sidebar

## How you can test it

- Run `jupyter-book build docs/` locally

### CODE REVIEW

File: docs/_toc.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Table of contents
+# Learn more at https://jupyterbook.org/customize/toc.html
+
+format: jb-book
+root: intro
+parts:
+- caption: Getting Started
+  chapters:
+    - title: Installation
+      file: installation
```
Comment (weiji14) [inline docs/_toc.yml:10]: What sections/subsections should we put on the sidebar? My initial ideas are:

- Getting started
  - Installation
- Data Preparation
  - Preparing the datacube
  - Preparing labels for evaluation model performance
- Running the model
  - Self-supervised learning
    - Linear Probing?
  - Generating embeddings

File: .github/workflows/deploy-docs.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
+# https://jupyterbook.org/en/stable/publish/gh-pages.html#automatically-host-your-book-with-github-actions
+name: deploy-book
+
+# Only run this when the main branch changes
+on:
+  # Uncomment the 'pull_request' line below to manually re-build Jupyter Book
+  pull_request:
```
Comment (weiji14) [inline .github/workflows/deploy-docs.yml:7]: Reminder to self: Comment this out before merging.
```suggestion
  # pull_request:
```

File: docs/_toc.yml (hunk 1 above)
Comment (yellowcap) [reply docs/_toc.yml:10]: - I would add a paragraph about the clay foundation and a link to the [linkedin page](https://www.linkedin.com/company/made-with-clay).
- We could add a section about how to run the datacube in batch

Comment (weiji14) [reply docs/_toc.yml:10]: Ok, added a link to the LinkedIn (and GitHub) page at commit bcbf080028c7067d658e3a4a758e7e9149a4e84d. Gonna merge this PR soon, so I'll open a separate issue to discuss about what sections/sub-sections to add later.

**Pull request merged**